ENGyn.NodesTestPlatform/ENGyn.NodesTestPlatform/Providers/CommandProvider.cs

```diff
@@ -0,0 +1,79 @@
+﻿using ENGyn.NodesTestPlatform.Commands;
+using ENGyn.NodesTestPlatform.Services;
+using ENGyn.NodesTestPlatform.Utils;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.IO;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace ENGyn.NodesTestPlatform.Providers
+{
+    public class CommandProvider : ICommandService
+    {
+        private readonly IReflectionService _reflectionService;
+        private readonly IConfigurationService _configurationService;
+        private string _currentExecutionDirectory;
+
+        public CommandProvider()
+        {
+            _reflectionService = new ReflectionProvider();
+            _configurationService = new ConfigurationProvider();
+            _currentExecutionDirectory = Environment.CurrentDirectory;
+        }
+
+        /// <summary>
+        /// Initializes a new node test platform project
+        /// </summary>
+        /// <param name="init">Init command instance</param>
+        public void Init(Init init)
+        {
+            bool projectNameIsValid = Regex.Match(init.ProjectName, @"^[^\s]+$").Success;
```

Comment:
Move the RegEx to a variable with a descriptive name.

Reply:
Done